system/src/system_cloud_internal.cpp

```diff
@@ -1104,8 +1104,13 @@ int spark_cloud_socket_connect()
     ip_address_error = determine_connection_address(ip_addr, port, server_addr, udp);
     if (!ip_address_error)
     {
-    		uint8_t local_port_offset = (PLATFORM_ID==3) ? 100 : 0;
-        sparkSocket = socket_create(AF_INET, udp ? SOCK_DGRAM : SOCK_STREAM, udp ? IPPROTO_UDP : IPPROTO_TCP, port+local_port_offset, NIF_DEFAULT);
+#if PLATFORM_ID == 3
```

Comment:
can we push this down to the HAL to avoid platform specifics in non-HAL code

Reply:
I'd disagree. In that case all UDP sockets would use ephemeral ports. For some specific use-cases we might need to use a specific UDP port.

We have something similar on Electron: https://github.com/particle-iot/firmware/blob/develop/system/src/system_task.cpp#L292

Might be a good idea to move this block there.

Reply:
What I mean is an option that lets the platform HAL decide the source port specifically for the cloud socket. Perhaps an option "UseEphemeralPortForCloud" that is sourced from the HAL, so there is no platform conditionality in the non-HAL code.